### PR TITLE
Convert integration AWS ES to gen (+ improve rules in gen format)

### PR DESCRIPTION
**Pull request by haedri**
feat : convert module integration_aws-elasticsearch to gen format
feat : convert module integration_aws-elasticache-common to gen format
feat : support for multiple signal in rules
feat : support for multiple time the same severity in rules
fixes https://github.com/claranet/terraform-signalfx-detectors/issues/516

### CODE REVIEW

File: modules/integration_aws-elasticache-common/conf/02-max-connection.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module: AWS ElastiCache
+name: "Max connection"
```
Comment (pdecat) [inline modules/integration_aws-elasticache-common/conf/02-max-connection.yaml:2]:
```suggestion
name: "Max connections"
```

File: modules/integration_aws-elasticsearch/conf/06-free-space.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module: AWS Elasticsearch
+name: "Free space"
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/06-free-space.yaml]:
```suggestion
name: "Free storage space"
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 3]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
 value_unit: "%"
 
 signals:
-  A:
+  error_stream:
     metric: 5xx
-  B:
+  open_search_stream:
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:14]:
```suggestion
  opensearch_stream:
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 4]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
...
     metric: OpenSearchRequests
-  signal:
-    formula: (A/B*100)
+  elastic_search_stream:
+    metric: ElasticsearchRequests
+  open_search_signal:
+    formula: (error_stream/open_search_stream*100)
+    publish: true
+  elastic_search_signal:
+    formula: (error_stream/elastic_search_stream*100)
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:22]:
```suggestion
  elasticsearch_stream:
    metric: ElasticsearchRequests
  opensearch_signal:
    formula: (error_stream/opensearch_stream*100)
    publish: true
  elasticsearch_signal:
    formula: (error_stream/elasticsearch_stream*100)
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 5]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
...
+    publish: true
 
 rules:
-  critical:
+  critical_open_search:
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:26]:
```suggestion
  opensearch_critical:
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 6]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
...
     threshold: 10
     comparator: ">"
     lasting_duration: "5m"
     lasting_at_least: 0.9
-  major:
+    signal: open_search_signal
+    severity: critical
+  major_open_search:
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:33]:
```suggestion
    signal: opensearch_signal
    severity: critical
  opensearch_major:
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 7]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
...
     threshold: 5
     comparator: ">"
-    dependency: critical
+    dependency: critical_open_search
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:36]:
```suggestion
    dependency: opensearch_critical
```

File: modules/integration_aws-elasticsearch/conf/03-5xx.yaml [hunk 8]
```diff
@@ -9,22 +9,47 @@ runbook_url: "https://docs.aws.amazon.com/opensearch-service/latest/developergui
...
+    lasting_at_least: 0.9
+    signal: elastic_search_signal
+    severity: critical
+  major_elastic_search:
+    threshold: 5
+    comparator: ">"
+    dependency: critical_elastic_search
+    lasting_duration: "5m"
+    lasting_at_least: 0.9
+    signal: elastic_search_signal
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/03-5xx.yaml:54]:
```suggestion
    signal: opensearch_signal
    severity: major
  elasticsearch_critical:
    threshold: 10
    comparator: ">"
    lasting_duration: "5m"
    lasting_at_least: 0.9
    signal: elasticsearch_signal
    severity: critical
  elasticsearch_major:
    threshold: 5
    comparator: ">"
    dependency: elasticsearch_critical
    lasting_duration: "5m"
    lasting_at_least: 0.9
    signal: elasticsearch_signal
```

File: modules/integration_aws-elasticsearch/conf/05-cluster-status.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+module: AWS Elasticsearch
+name: Cluster Status
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/05-cluster-status.yaml:2]:
```suggestion
name: Cluster status
```

File: modules/integration_aws-elasticsearch/conf/07-ultrawarm-free-space.yaml [hunk 10]
```diff
@@ -0,0 +1,21 @@
+module: AWS Elasticsearch
+name: "Ultrawarm Free space"
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/07-ultrawarm-free-space.yaml]:
```suggestion
name: "Ultrawarm free storage space"
```

File: modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml [hunk 11]
```diff
@@ -0,0 +1,24 @@
+module: AWS Elasticsearch
+name: "Cluster CPU"
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml:2]:
```suggestion
name: "CPU utilization"
```

File: modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+aggregation: ""
+filtering: "filter('namespace', 'AWS/ES') and filter('stat', 'upper') and filter('NodeId', '*')"
+
+signals:
+  data_node_cpu:
+    metric: CPUUtilization
+  warm_node_cpu:
+    metric: WarmCPUUtilization
+  signal:
+    formula: union(data_node_cpu, warm_node_cpu)
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml:14]: We should probably process those signals independently as Ultrawarm nodes do not have the same profile as normal nodes.

File: modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+aggregation: ""
+filtering: "filter('namespace', 'AWS/ES') and filter('stat', 'upper') and filter('NodeId', '*')"
+
+signals:
+  signal:
+    metric: MasterCPUUtilization
+
+rules:
+  major:
+    threshold: 50
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml:14]: AWS [recommends](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-cloudwatchmetrics.html#managedomains-cloudwatchmetrics-master-node-metrics) _increasing the size of the instance type when this metric reaches 60 percent_.

File: modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml [hunk 14]
```diff
@@ -0,0 +1,19 @@
+module: AWS Elasticsearch
+name: "Master CPU"
```
Comment (pdecat) [inline modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml:2]:
```suggestion
name: "Master CPU utilization"
```

**Review by pdecat — COMMENTED** [review]
Looks good overall! Made a few suggestions that may involve also renaming files, inputs and/or outputs.

File: modules/integration_aws-elasticache-common/conf/02-max-connection.yaml (hunk 1 above)
Comment (haedri) [reply modules/integration_aws-elasticache-common/conf/02-max-connection.yaml]: I put back the previous name

File: modules/integration_aws-elasticache-common/conf/06-evictions-growing.yaml [hunk 15]
```diff
@@ -0,0 +1,21 @@
+module: AWS ElastiCache
+name: "Evictions growing"
```
[the remark this answers is not in the document]
Comment (haedri) [reply modules/integration_aws-elasticache-common/conf/06-evictions-growing.yaml]: I put back the previous name

File: modules/integration_aws-elasticache-common/conf/05-free-memory.yaml [hunk 16]
```diff
@@ -0,0 +1,21 @@
+module: AWS ElastiCache
+name: "Free memory"
```
[the remark this answers is not in the document]
Comment (haedri) [reply modules/integration_aws-elasticache-common/conf/05-free-memory.yaml]: Done

File: modules/integration_aws-elasticache-common/conf/04-swap.yaml [hunk 17]
```diff
@@ -0,0 +1,21 @@
+module: AWS ElastiCache
+name: "Swap"
```
[the remark this answers is not in the document]
Comment (haedri) [reply modules/integration_aws-elasticache-common/conf/04-swap.yaml]: Done

File: modules/integration_aws-elasticsearch/README.md [hunk 18]
```diff
@@ -110,6 +110,7 @@ Here is the list of required metrics for detectors in this module.
 * `ClusterStatus.red`
 * `ClusterStatus.yellow`
 * `CPUUtilization`
+* `ElasticsearchRequests`
```
[the remark this answers is not in the document]
Comment (haedri) [reply modules/integration_aws-elasticsearch/README.md:114]: Done

File: modules/integration_aws-elasticsearch/conf/06-free-space.yaml (hunk 2 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/06-free-space.yaml]: Done

File: modules/integration_aws-elasticsearch/conf/07-ultrawarm-free-space.yaml (hunk 10 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/07-ultrawarm-free-space.yaml]: Done

File: modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml (hunk 11 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml]: Done

File: modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml (hunk 12 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/08-cluster-cpu.yaml:15]: It's a bit too much of a change for this PR, maybe for another one

File: modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml (hunk 13 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml]: Done

File: modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml (hunk 14 above)
Comment (haedri) [reply modules/integration_aws-elasticsearch/conf/09-master-cpu.yaml]: Done

**Review by JMHAUTBOIS-Claranet — APPROVED** [review]

**Pull request merged**